### PR TITLE
Shank 389 facilities integration tests

### PR DESCRIPTION
**Pull request by ilaflamme-va**
Integration Tests pointed at vets-api facilities endpoints.

- supports multiple accept headers (media types)
  - each test makes multiple requests
- No output validation, only status code
- They do NOT run automatically (@ignore on the IT.java)
- run them by providing the  `-Dsentinel.facilities.url=<url>` and the `-Dvets-api-facilities-apikey`

### CODE REVIEW

File: facilities-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <version>${docker-maven-plugin.version}</version>
+            <executions>
+              <execution>
+                <id>build-test</id>
+                <configuration>
+                  <images>
+                    <image>
+                      <name>${docker.organization}/lighthouse-facilities-cdw-tests</name>
```
Comment (eclendenning-va) [inline facilities-tests/pom.xml]: not cdw,  just `lighthouse-facilities-tests`

File: facilities-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                <goals>
+                  <goal>build</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>deploy-test</id>
+                <configuration>
+                  <images>
+                    <image>
+                      <name>${docker.organization}/lighthouse-facilities-cdw-tests</name>
```
Comment (eclendenning-va) [inline facilities-tests/pom.xml]: same here

File: facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/TestIds.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package gov.va.api.lighthouse.facilities.tests;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+@Builder(toBuilder = true)
+public class TestIds {
```
Comment (jhulbert-va) [inline facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/TestIds.java]: `FacilitiesIds`?

File: facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      case STAGING_LAB:
+      case LAB:
+      case LOCAL:
+        return local();
+      default:
+        throw new IllegalArgumentException("Unknown sentinel environment: " + Environment.get());
+    }
+  }
+
+  private TestIds facilitiesIds() {
```
Comment (jhulbert-va) [inline facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java]: `localFacilitiesIds`? I assume we'll need different Ids per environment.

File: facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/TestIds.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  @NonNull String facility;
+  @NonNull String facilities;
```
Comment (jhulbert-va) [inline facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/TestIds.java]: These could probably be made clearer. I had to do a double take on them before understanding it was for searches vs reads.

File: facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java (hunk 4 above)
Comment (ilaflamme-va) [reply facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java]: I don't think so. Facility Data doesn't change... unless we plan to synthesize it for lab

**Comment by jblefkove-va** [comment]
Agreed, facilities IDs won't change between environments

File: facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/FacilitiesIT.java [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+@Ignore
+@Slf4j
+public class FacilitiesIT {
+  @Test
+  @Category({AllFacilities.class})
+  public void all() {
+    final String request = "v0/facilities/all";
+    ExpectedResponse.of(makeRequest("application/vnd.geo+json", request)).expect(200);
+    ExpectedResponse.of(makeRequest("text/csv", request)).expect(200);
```
[the remark this answers is not in the document]
Comment (ilaflamme-va) [reply facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/FacilitiesIT.java]: :+1:

File: facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/FacilitiesIT.java [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+  public void searchByBoundingBox() {
+    final String bbox = systemDefinition().testIds().bbox();
+    final String request = "v0/facilities?" + bbox;
+    ExpectedResponse.of(makeRequest("application/vnd.geo+json", request)).expect(200);
+    ExpectedResponse.of(makeRequest("application/json", request)).expect(200);
+  }
+
+  @Test
+  @Category(SearchByBoundingBox.class)
+  public void searchByBoundingBoxWithServices() {
```
[the remark this answers is not in the document]
Comment (ilaflamme-va) [reply facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/FacilitiesIT.java]: :+1:

File: facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    return TestIds.builder()
+        .facility("vba_322b")
+        .facilities("nca_055,nca_s1001,vba_322b,vc_0101V,vha_402GA")
+        .latitude("28.112464")
+        .longitude("-80.7015994")
+        .bbox("bbox[]=-81.47&bbox[]=27.48&bbox[]=-79.97&bbox[]=28.98")
```
[the remark this answers is not in the document]
Comment (ilaflamme-va) [reply facilities-tests/src/main/java/gov/va/api/lighthouse/facilities/tests/SystemDefinitions.java]: I'm not sure, I thought the string was the easiest to work with at the time.

**Pull request merged**